Binary Testing Framework Outputs Raw tfjson.Plan During "plan was not empty"

### SDK version

```
v1.12.0
```

### Relevant provider source code

```go
func TestAccAWSEbsSnapshotCopy_tags(t *testing.T) {
	var snapshot ec2.Snapshot
	resourceName := "aws_ebs_snapshot_copy.test"

	resource.ParallelTest(t, resource.TestCase{
		PreCheck:     func() { testAccPreCheck(t) },
		Providers:    testAccProviders,
		CheckDestroy: testAccCheckEbsSnapshotCopyDestroy,
		Steps: []resource.TestStep{
			{
				Config: testAccAwsEbsSnapshotCopyConfigTags1("key1", "value1"),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckEbsSnapshotCopyExists(resourceName, &snapshot),
					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1"),
				),
			},
			{
				Config: testAccAwsEbsSnapshotCopyConfigTags2("key1", "value1updated", "key2", "value2"),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckEbsSnapshotCopyExists(resourceName, &snapshot),
					resource.TestCheckResourceAttr(resourceName, "tags.%", "3"),
					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1updated"),
					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
				),
			},
			{
				Config: testAccAwsEbsSnapshotCopyConfigTags1("key2", "value2"),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckEbsSnapshotCopyExists(resourceName, &snapshot),
					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
				),
			},
		},
	})
}
```

### Terraform Configuration Files

```hcl
func testAccAwsEbsSnapshotCopyConfigTags1(tagKey1, tagValue1 string) string {
	return fmt.Sprintf(`
data "aws_availability_zones" "available" {
  state = "available"

  filter {
    name   = "opt-in-status"
    values = ["opt-in-not-required"]
  }
}

data "aws_region" "current" {}

resource "aws_ebs_volume" "test" {
  availability_zone = "${data.aws_availability_zones.available.names[0]}"
  size              = 1
}

resource "aws_ebs_snapshot" "test" {
  volume_id = "${aws_ebs_volume.test.id}"

  tags = {
    Name = "testAccAwsEbsSnapshotCopyConfig"
  }
}

resource "aws_ebs_snapshot_copy" "test" {
  source_snapshot_id = "${aws_ebs_snapshot.test.id}"
  source_region      = "${data.aws_region.current.name}"

  tags = {
    Name = "testAccAwsEbsSnapshotCopyConfig"
	"%s" = "%s"
  }
}
`, tagKey1, tagValue1)
}

func testAccAwsEbsSnapshotCopyConfigTags2(tagKey1, tagValue1, tagKey2, tagValue2 string) string {
	return fmt.Sprintf(`
data "aws_availability_zones" "available" {
  state = "available"

  filter {
    name   = "opt-in-status"
    values = ["opt-in-not-required"]
  }
}

data "aws_region" "current" {}

resource "aws_ebs_volume" "test" {
  availability_zone = "${data.aws_availability_zones.available.names[0]}"
  size              = 1
}

resource "aws_ebs_snapshot" "test" {
  volume_id = "${aws_ebs_volume.test.id}"

  tags = {
    Name = "testAccAwsEbsSnapshotCopyConfig"
  }
}

resource "aws_ebs_snapshot_copy" "test" {
  source_snapshot_id = "${aws_ebs_snapshot.test.id}"
  source_region      = "${data.aws_region.current.name}"

  tags = {
    Name = "testAccAwsEbsSnapshotCopyConfig"
	"%s" = "%s"
	"%s" = "%s"
  }
}
`, tagKey1, tagValue1, tagKey2, tagValue2)
}
```

### Debug Output

[TeamCity link](https://OMITTED/repository/download/Aws_ProviderAwsCommercialPartitionMainKmoeTest/112470:id/debug-aws-554efd4-TestAccAWSEbsSnapshotCopy_tags.log) (replace OMITTED with correct TeamCity hostname)

### Expected Behavior

```
--- FAIL: TestAccAWSEbsSnapshotCopy_tags (58.67s)
testing_new_config.go:53: After applying this test step, the plan was not empty.

(Terraform 0.12 equivalent plan difference output)
```

### Actual Behavior

```
--- FAIL: TestAccAWSEbsSnapshotCopy_tags (58.67s)
testing_new_config.go:53: After applying this test step, the plan was not empty. (*tfjson.Plan)(0xc0008d0ea0)({
FormatVersion: (string) (len=3) "0.1",
TerraformVersion: (string) (len=7) "0.12.20",
Variables: (map[string]*tfjson.PlanVariable) <nil>,
PlannedValues: (*tfjson.StateValues)(0xc000b47aa0)({
Outputs: (map[string]*tfjson.StateOutput) <nil>,
RootModule: (*tfjson.StateModule)(0xc000aefcc0)({
Resources: ([]*tfjson.StateResource) (len=3 cap=4) {
(*tfjson.StateResource)(0xc000ba8e60)({
... a lot more! ...
```

### Steps to Reproduce

1. `TF_ACC=1 go test ./aws -v -run='TestAccAWSEbsSnapshotCopy_tags'` (although this error may be inconsistent for this particular test)

--- a/go.mod
+++ b/go.mod
@@ -27,7 +27,7 @@ require (
 	github.com/hashicorp/hcl/v2 v2.3.0
 	github.com/hashicorp/logutils v1.0.0
 	github.com/hashicorp/terraform-json v0.5.0
-	github.com/hashicorp/terraform-plugin-test/v2 v2.0.0-20200724200815-faa9931ac59e
+	github.com/hashicorp/terraform-plugin-test/v2 v2.0.0-20200807213038-0c54794f0aee
 	github.com/hashicorp/yamux v0.0.0-20181012175058-2f1d1f20f75d // indirect
 	github.com/keybase/go-crypto v0.0.0-20161004153544-93f5b35093ba
 	github.com/kylelemons/godebug v1.1.0 // indirect

--- a/go.mod
+++ b/go.mod
@@ -27,7 +27,7 @@ require (
 	github.com/hashicorp/hcl/v2 v2.3.0
 	github.com/hashicorp/logutils v1.0.0
 	github.com/hashicorp/terraform-json v0.5.0
-	github.com/hashicorp/terraform-plugin-test/v2 v2.0.0-20200807213038-0c54794f0aee
+	github.com/hashicorp/terraform-plugin-test/v2 v2.0.0
 	github.com/hashicorp/yamux v0.0.0-20181012175058-2f1d1f20f75d // indirect
 	github.com/keybase/go-crypto v0.0.0-20161004153544-93f5b35093ba
 	github.com/kylelemons/godebug v1.1.0 // indirect

--- a/helper/resource/testing_new_config.go
+++ b/helper/resource/testing_new_config.go
@@ -1,7 +1,6 @@
 package resource
 
 import (
-	"github.com/davecgh/go-spew/spew"
 	tfjson "github.com/hashicorp/terraform-json"
 	tftest "github.com/hashicorp/terraform-plugin-test/v2"
 	testing "github.com/mitchellh/go-testing-interface"
@@ -11,9 +10,6 @@ import (
 
 func testStepNewConfig(t testing.T, c TestCase, wd *tftest.WorkingDir, step TestStep) error {
 	t.Helper()
-
-	spewConf := spew.NewDefaultConfig()
-	spewConf.SortKeys = true
 
 	var idRefreshCheck *terraform.ResourceState
 	idRefresh := c.IDRefreshName != ""
@@ -90,8 +86,15 @@ func testStepNewConfig(t testing.T, c TestCase, wd *tftest.WorkingDir, step Test
 		if step.ExpectNonEmptyPlan {
 			t.Log("[INFO] Got non-empty plan, as expected")
 		} else {
-
-			t.Fatalf("After applying this test step, the plan was not empty. %s", spewConf.Sdump(plan))
+			var stdout string
+			err = runProviderCommand(t, func() error {
+				stdout = wd.RequireSavedPlanRaw(t)
+				return nil
+			}, wd, c.ProviderFactories)
+			if err != nil {
+				return err
+			}
+			t.Fatalf("After applying this test step, the plan was not empty.\nstdout:\n\n%s", stdout)
 		}
 	}
 
@@ -126,8 +129,15 @@ func testStepNewConfig(t testing.T, c TestCase, wd *tftest.WorkingDir, step Test
 		if step.ExpectNonEmptyPlan {
 			t.Log("[INFO] Got non-empty plan, as expected")
 		} else {
-
-			t.Fatalf("After applying this test step and performing a `terraform refresh`, the plan was not empty. %s", spewConf.Sdump(plan))
+			var stdout string
+			err = runProviderCommand(t, func() error {
+				stdout = wd.RequireSavedPlanRaw(t)
+				return nil
+			}, wd, c.ProviderFactories)
+			if err != nil {
+				return err
+			}
+			t.Fatalf("After applying this test step and performing a `terraform refresh`, the plan was not empty.\nstdout\n\n%s", stdout)
 		}
 	}
 

--- a/helper/resource/testing_new_config.go
+++ b/helper/resource/testing_new_config.go
@@ -88,7 +88,7 @@ func testStepNewConfig(t testing.T, c TestCase, wd *tftest.WorkingDir, step Test
 		} else {
 			var stdout string
 			err = runProviderCommand(t, func() error {
-				stdout = wd.RequireSavedPlanRaw(t)
+				stdout = wd.RequireSavedPlanStdout(t)
 				return nil
 			}, wd, c.ProviderFactories)
 			if err != nil {
@@ -131,7 +131,7 @@ func testStepNewConfig(t testing.T, c TestCase, wd *tftest.WorkingDir, step Test
 		} else {
 			var stdout string
 			err = runProviderCommand(t, func() error {
-				stdout = wd.RequireSavedPlanRaw(t)
+				stdout = wd.RequireSavedPlanStdout(t)
 				return nil
 			}, wd, c.ProviderFactories)
 			if err != nil {

